kf/processCheck.go

```diff
@@ -86,3 +87,63 @@ func (c *pCheck) pMonitorWorker(bpfProgs map[string]*list.List, direction string
 		}
 	}
 }
+
+func (c *pCheck) pMonitorProbeWorker(bpfProgs *list.List) {
+	for range time.NewTicker(c.retryMonitorDelay).C {
+		if bpfProgs == nil {
+			time.Sleep(time.Second)
+			continue
+		}
+		for e := bpfProgs.Front(); e != nil; e = e.Next() {
+			bpf := e.Value.(*BPF)
+			if bpf.Program.AdminStatus == models.Disabled {
+				continue
+			}
+			userProgram, bpfProgram, _ := bpf.isRunning()
+			if userProgram && bpfProgram {
+				stats.SetWithVersion(1.0, stats.BPFRunning, bpf.Program.Name, bpf.Program.Version, "", "")
+				continue
+			}
+			// Not running trying to restart
+			if bpf.RestartCount < c.MaxRetryCount && bpf.Program.AdminStatus == models.Enabled {
```

Comment:
Should there be some sort of exponential backoff if restarts keep failing? Or is MaxRetryCount usually small?

Reply:
Default max retry count is 3